File: datamodel/src/main/java/datamodel/clinical/treatment/TreatmentCategory.java
package datamodel.clinical.treatment;

import org.jetbrains.annotations.NotNull;

public enum TreatmentCategory {
    CHEMOTHERAPY(true),
    RADIOTHERAPY(true),
    CHEMORADIOTHERAPY(false),
    TARGETED_THERAPY(true),
    IMMUNOTHERAPY(true),
    HORMONE_THERAPY(true),
    ANTIVIRAL_THERAPY(false),
    SUPPORTIVE_TREATMENT(true),
    SURGERY(false),
    TRANSPLANTATION(true),
    TRIAL(true),
    CAR_T(true),
    TCR_T(false),
    GENE_THERAPY(false),
    PROPHYLACTIC_TREATMENT(false),
    ABLATION(true);

    private final boolean hasType;

    TreatmentCategory(final boolean hasType) {
        this.hasType = hasType;
    }

    public boolean hasType() {
        return hasType;
    }
}

File: algo/src/main/kotlin/com/hartwig/actin/datamodel/clinical/treatment/TreatmentCategory.kt
package com.hartwig.actin.datamodel.clinical.treatment

import com.hartwig.actin.datamodel.Displayable

enum class TreatmentCategory : Displayable {
    CHEMOTHERAPY,
    TARGETED_THERAPY,
    IMMUNOTHERAPY,
    HORMONE_THERAPY,
    SURGERY,
    TRANSPLANTATION,
    RADIOTHERAPY,
    ABLATION,
    SUPPORTIVE_TREATMENT;

    override fun display(): String {
        return this.toString().replace("_", " ").lowercase()
    }

    companion object {
        val SYSTEMIC_CANCER_TREATMENT_CATEGORIES: Set<TreatmentCategory> =
            setOf(CHEMOTHERAPY, TARGETED_THERAPY, IMMUNOTHERAPY, HORMONE_THERAPY)
    }
}
